crates/locutus-node/src/node/test_utils.rs

```diff
@@ -176,12 +185,38 @@ impl SimNetwork {
         }
     }
 
+    pub fn build_with_specs(&mut self, net_specs: HashMap<String, NodeSpecs>) {
```

Comment:
looks good but we can have a sinle function; initialize_peer takes an Option<NodeSpec> instead and you do initialize MemoryEventGen based on it being None or Some(_);

You can also use remove and t